### PR TITLE
Send notifications when scan results arrive

### PR DESCRIPTION
**Pull request by prasanthu**
TODO: We need to fine tune the messages.

### CODE REVIEW

File: app-notifier/src/main/kotlin/org/ostelco/prime/appnotifier/FirebaseAppNotifier.kt [hunk 1]
```diff
@@ -19,10 +20,15 @@ class FirebaseAppNotifier: AppNotifier {
 
     override fun notify(customerId: String, title: String, body: String) {
         println("Will try to notify customer with Id : $customerId")
-        sendNotification(customerId, title, body)
+        sendNotification(customerId, title, body, data = null)
     }
 
-    private fun sendNotification(customerId: String, title: String, body: String) {
+    override fun notify(customerId: String, title: String, body: String, data: Map<String, String>) {
+        println("Will try to notify-with-data customer with Id : $customerId")
```
Comment (mpeterss) [inline app-notifier/src/main/kotlin/org/ostelco/prime/appnotifier/FirebaseAppNotifier.kt]: We should change this to use logger and not println. I have no idea why I did that in the first place :)

File: neo4j-store/src/main/kotlin/org/ostelco/prime/storage/graph/Neo4jStore.kt [hunk 2]
```diff
@@ -1133,24 +1134,39 @@ object Neo4jStoreSingleton : GraphStore {
...
-
+                val extendedStatus = scanInformationDatastore.getExtendedStatusInformation(vendorData)
                 if (scanInformation.status == ScanStatus.APPROVED) {
 
                     logger.info("Inserting scan Information to cloud storage : id: ${scanInformation.scanId} countryCode: ${scanInformation.countryCode}")
                     scanInformationDatastore.upsertVendorScanInformation(customer.id, scanInformation.countryCode, vendorData)
                             .flatMap {
+                                appNotifier.notify(
+                                        customerId = customer.id,
+                                        title = "eKYC Status",
```
Comment (mpeterss) [inline neo4j-store/src/main/kotlin/org/ostelco/prime/storage/graph/Neo4jStore.kt]: Should we move to constant for the common strings?

https://blog.egorand.me/where-do-i-put-my-constants-in-kotlin/

File: scaninfo-datastore/src/main/kotlin/org/ostelco/prime/storage/scaninfo/ScanInfoStore.kt [hunk 3]
```diff
@@ -373,6 +377,50 @@ object JumioHelper {
...
+        val verificationStatus: String = vendorData.getFirst(JumioScanData.VERIFICATION_STATUS.s)
+        val identityVerificationData: String? = vendorData.getFirst(JumioScanData.IDENTITY_VERIFICATION.s)
+
+        extendedStatus.putIfAbsent(JumioScanData.VERIFICATION_STATUS.s, verificationStatus)
+        if (verificationStatus.toUpperCase() == JumioScanData.APPROVED_VERIFIED.s) {
+            val identityVerification = toRegularMap(identityVerificationData)
+            if (identityVerification == null) {
+                extendedStatus.putIfAbsent(JumioScanData.REJECT_REASON.s, JumioScanData.PRIME_MISSING_IDENTITY_VERIFICATION.s)
+            } else {
+                // identityVerification field is present
```
Comment (mpeterss) [inline scaninfo-datastore/src/main/kotlin/org/ostelco/prime/storage/scaninfo/ScanInfoStore.kt]: Refactor this out a bit:

private fun addIdentityVerification () {

    val similarity = identityVerification[JumioScanData.SIMILARITY.s]
    val validity = identityVerification[JumioScanData.VALIDITY.s]
    val reason = identityVerification[JumioScanData.REASON.s]

    if (similarity == null || validity == null) {
        extendedStatus.putIfAbsent(JumioScanData.IDENTITY_VERIFICATION.s, JumioScanData.PRIME_IDENTITY_VERIFICATION_FAILED.s)
    } else if (similarity.toUpperCase() == JumioScanData.MATCH.s && validity.toUpperCase() == JumioScanData.TRUE.s) {
        extendedStatus.putIfAbsent(JumioScanData.IDENTITY_VERIFICATION.s, JumioScanData.PRIME_IDENTITY_VALID_SIMILAR.s)
    }

    if (similarity.toUpperCase() != JumioScanData.MATCH.s) {
        extendedStatus.putIfAbsent(JumioScanData.REJECT_REASON.s, similarity)
    } else if (validity.toUpperCase() != JumioScanData.TRUE.s) {
        extendedStatus.putIfAbsent(JumioScanData.REJECT_REASON.s, reason ?: JumioScanData.PRIME_MISSING_IDENTITY_REASON.s)
    }
}

File: app-notifier/src/main/kotlin/org/ostelco/prime/appnotifier/FirebaseAppNotifier.kt (hunk 1 above)
Comment (prasanthu) [reply app-notifier/src/main/kotlin/org/ostelco/prime/appnotifier/FirebaseAppNotifier.kt]: done

File: neo4j-store/src/main/kotlin/org/ostelco/prime/storage/graph/Neo4jStore.kt (hunk 2 above)
Comment (prasanthu) [reply neo4j-store/src/main/kotlin/org/ostelco/prime/storage/graph/Neo4jStore.kt]: We are implementing consts in a different way

File: scaninfo-datastore/src/main/kotlin/org/ostelco/prime/storage/scaninfo/ScanInfoStore.kt (hunk 3 above)
Comment (prasanthu) [reply scaninfo-datastore/src/main/kotlin/org/ostelco/prime/storage/scaninfo/ScanInfoStore.kt]: done

**Pull request merged**